src/cms/src/api/park-contact/content-types/park-contact/lifecycles.js

```diff
@@ -6,12 +6,24 @@ const updateName = async (data, where) => {
         const parkContact = await strapi.entityService.findOne(
             "api::park-contact.park-contact", id, { populate: '*' }
         )
-        const protectedArea = parkContact.protectedArea
+        let protectedArea = parkContact.protectedArea
         const parkOperatorContact = parkContact.parkOperatorContact
 
+        // Check if new protectedArea is being added
+        if (data?.protectedArea?.connect?.length > 0) {
+          protectedArea = await strapi.entityService.findOne(
+            "api::protected-area.protected-area", data?.protectedArea.connect[0].id
+          )
+        // Check if current protectedArea is being removed
+        } else if (data?.protectedArea?.disconnect?.length > 0) {
+          protectedArea = { orcs: 0 }
+        }
+
         data.name = ""
         if (protectedArea) {
             data.name = protectedArea.orcs
+        } else {
+            data.name = 0
         }
```

Comment:
should this be checking the orcs? If disconnect.length > 0 and you set the orcs to 0, `protectedArea` will still evaluate to true in this check. Just making sure that's intended.

Reply:
Yes, that's intended. If `parkContact` is created for the first time without `protectedArea`, it'll fall in this `else {}`. However once users `protectedArea` add to `parkContact` and remove `protectedArea` from it, and click the Save button, it'll still return the previous `protectedArea` (I guess it's because of how Strapi lifecycle works) 😅 so I needed to change its `orcs` when it's removed (disconnected).

Reply:
Thanks for clarifying!